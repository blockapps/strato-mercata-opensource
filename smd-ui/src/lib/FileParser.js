export function getImportStatements(source) {
    const splittedSource = source.split("\n")
    // TODO: This does not guard against multiline strings or comments
    const newImports = splittedSource.filter((line) => /^\s*import /.test(line))
    const newFileList = []
    newImports.map((item) => {
        const splittedItem = item.split(' ')
        const filename = splittedItem.filter((word) => word.includes('.sol'))
        newFileList.push(filename.length > 0 && filename[0].replace(/['";]+/g, ''))
        return item
    })
    return newFileList
}

export function replaceImportStatementsWithSource(source, importsCode, tabData) {
    const splittedSource = source.split("\n")
    const filteredList = splittedSource.map((item) => {
        const code = importsCode.filter(
            (codeItem) =>
                item.includes(codeItem.title)
        )
        return code.length > 0 ? getFileAndReplaceWithImport(code[0].text, tabData) : item }
    )
    filteredList.join("\r\n")
    return filteredList
}

export function getFileAndReplaceWithImport(code, tabData) {
    const newFileList = getImportStatements(code)
    if (newFileList.length === 0) {
        return code
    }
    const items = tabData.filter((item) =>
        newFileList.includes(item.title)
    )
    const replacedCode = replaceImportStatementsWithSource(code, items, tabData)
    return replacedCode.join("\r\n")
}
