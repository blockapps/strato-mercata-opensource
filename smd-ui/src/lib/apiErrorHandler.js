export function handleApiError(response) {
  return new Promise(function(resolve,reject){
    if(response.error) {
      reject(response.error);
    }
    else {
      resolve(response);
    }
  })
}
