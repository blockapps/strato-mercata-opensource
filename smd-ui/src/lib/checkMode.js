import { env } from "../env";

export function isOauthEnabled() {
  return env.OAUTH_ENABLED
}