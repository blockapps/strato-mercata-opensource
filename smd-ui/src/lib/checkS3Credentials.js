import { env } from "../env";

export function isS3Available() {
  return env.EXT_STORAGE_ENABLED === 'true'
}
