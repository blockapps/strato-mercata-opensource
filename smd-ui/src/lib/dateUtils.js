import moment from 'moment';

export function parseDateFromString(timeAsString) {
  const utcDate = timeAsString.replace(' UTC', 'Z')
  const formattedDate = moment(utcDate).format('YYYY-MM-DD hh:mm:ss A')
  return formattedDate
}

export function parseDateFromTimestamp(timestamp) {
  return parseDateFromString(moment.unix(timestamp).utc().toString());
}
