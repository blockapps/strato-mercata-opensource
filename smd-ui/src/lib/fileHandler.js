export function downloadFile(name, content) {
    var element = document.createElement('a');
    element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(content));
    element.setAttribute('download', name);
    element.style.display = 'none';
    document.body.appendChild(element);
    element.click();
    document.body.removeChild(element);
}

export function downloadPDFFile(name, content) {
    var element = document.createElement('a');
    element.setAttribute('href', content);
    element.setAttribute('download', name);
    element.style.display = 'none';
    document.body.appendChild(element);
    element.click();
    document.body.removeChild(element);
}