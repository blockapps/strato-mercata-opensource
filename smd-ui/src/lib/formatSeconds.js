import React from "react";
export function sec2Date(sec){
    if (sec === undefined || sec <= 0 ){
        return <div></div>;
    }
    let seconds = sec;
    const days = Math.floor(seconds / (3600*24));
    const daysString = days.toString().padStart(2, "0");
    seconds  -= days*3600*24;
    const hrs = Math.floor(seconds / 3600);
    const hoursString = hrs.toString().padStart(2, "0");
    seconds  -= hrs*3600;
    const mnts = Math.floor(seconds / 60);
    const minutesString = mnts.toString().padStart(2, "0");
    seconds  -= mnts*60;
    const secondsString = Math.floor(seconds).toString().padStart(2, "0");
    const ret = <div>
        Uptime {days > 0 ? <span>{daysString}d:</span> : null}
         <span>
             {hoursString}:
        </span>
        <span>
             {minutesString}:

        </span>
        <span>
             {secondsString}
        </span>
    </div>
    return ret;
}
