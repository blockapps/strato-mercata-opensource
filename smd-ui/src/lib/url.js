/*
  This will create the URL with the replace of params and create and append the query
  options should contain the query and params parameter.
  Also, this one remove the complexity of chainId. null values will be neglected will not be appended to query
*/
export function createUrl(url, options = {}) {
  const { params, query } = options;

  const withParams = params ? url.replace(/::(\w+)/g, (_, key) => {
    return params[key];
  }) : url;

  return query ? createQuery(withParams, query) : withParams;
}

// query
function createQuery(url, query) {
  const keys = Object.keys(query);

  keys.forEach((key, index) => {
    if (query[key]) {
      if (url.indexOf("?") < 0) {
        url += '?';
      } else {
        url += '&';
      }

      url += `${key}=${query[key]}`;
    }
  });

  return url;
}
