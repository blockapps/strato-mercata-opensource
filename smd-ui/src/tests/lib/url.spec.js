import { createUrl } from '../../lib/url';

describe('Lib: url', () => {

  test('create url', () => {
    const baseUrl = 'http://localhost/user/::username/::address';
    const options = { params: { username: 'clinicalops', address: '0X22255555' }, query: { chainid: '54216454215454', resolve: true } };
    expect(createUrl(baseUrl, options)).toMatchSnapshot();
  });

  test('create url (query only)', () => {
    const baseUrl = 'http://localhost/user';
    const options = { query: { chainid: '54216454215454', resolve: true } };
    expect(createUrl(baseUrl, options)).toMatchSnapshot();
  });

  test('create url (params only)', () => {
    const baseUrl = 'http://localhost/user/::username/::address';
    const options = { params: { username: 'clinicalops', address: '0X22255555' } };
    expect(createUrl(baseUrl, options)).toMatchSnapshot();
  });

  test('create url (null values should not be append)', () => {
    /*
     HERE: If chainid is null that will not be appended as a query
     Also: resolve: false will not be append as a query
    */
    const baseUrl = 'http://localhost/user/::username/::address';
    const options = { params: { username: 'clinicalops', address: '0X22255555' }, query: { chainid: null, resolve: false } };
    expect(createUrl(baseUrl, options)).toMatchSnapshot();
  });

  test('Empty options', () => {
    const baseUrl = 'http://localhost/user';
    expect(createUrl(baseUrl)).toMatchSnapshot();
  });

});
