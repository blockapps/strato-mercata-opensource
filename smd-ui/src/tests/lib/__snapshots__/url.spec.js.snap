// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lib: url Empty options 1`] = `"http://localhost/user"`;

exports[`Lib: url create url (null values should not be append) 1`] = `"http://localhost/user/clinicalops/0X22255555"`;

exports[`Lib: url create url (params only) 1`] = `"http://localhost/user/clinicalops/0X22255555"`;

exports[`Lib: url create url (query only) 1`] = `"http://localhost/user?chainid=54216454215454&resolve=true"`;

exports[`Lib: url create url 1`] = `"http://localhost/user/clinicalops/0X22255555?chainid=54216454215454&resolve=true"`;
